power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -53,14 +198,62 @@ inline void get_component_sequence(MainModelState<ComponentContainer> const& sta
 template <component_c Component, class ComponentContainer,
           forward_iterator_like<typename Component::UpdateType> ForwardIterator>
     requires model_component_state_c<MainModelState, ComponentContainer, Component>
-inline std::vector<Idx2D> get_component_sequence(MainModelState<ComponentContainer> const& state, ForwardIterator begin,
-                                                 ForwardIterator end, Idx n_comp_elements = na_Idx) {
+inline std::vector<Idx2D> get_component_sequence_by_it(MainModelState<ComponentContainer> const& state,
```

Comment:
Please rename this to `get_component_sequence_by_iter` / `get_component_sequence_by_range` or anything that does not remotely imply a terrifying figure...

Reply:
I got you!